woocommerce/woocommerce
# Delete old marketing recommendations transient upon update to WC 8.6.0

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR depends on PR https://github.com/woocommerce/woocommerce/pull/43001, and it assumes that both PR would be released in WooCommerce 8.6.0.

In this PR, we add database update code so that upon updating WooCommerce to version 8.6.0, we remove the old marketing recommendations transient `wc_marketing_recommended_plugins` as it is not used anymore. It is superseded by a new transient `woocommerce_admin_marketing_recommendations_specs` that is created by `MarketingRecommendationsDataSourcePoller` introduced in PR https://github.com/woocommerce/woocommerce/pull/43001.



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Setup a test site with old WooCommerce version, e.g. https://github.com/woocommerce/woocommerce/releases/tag/8.4.0.
2. The marketplace suggestions feature should be turned on in `/wp-admin/admin.php?page=wc-settings&tab=advanced&section=woocommerce_com`
3. In the test site, visit the Marketing page: `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`
4. There should be a transient called `wc_marketing_recommended_plugins`. You can view this using a plugin called Transients Manager.
5. Check out this PR branch, and build the woocommerce.zip file using `pnpm --filter=@woocommerce/plugin-woocommerce run build:zip`.
6. Update WooCommerce using the new woocommerce.zip plugin file.
7. You should see a "WooCommerce database update required" notice. Go ahead to update the database, and wait for it to finish.
8. The `wc_marketing_recommended_plugins` transient should be deleted by the database update process.
9. Visit the Marketing page: `/wp-admin/admin.php?page=wc-admin&path=%2Fmarketing`
10. There should be a new transient now called `woocommerce_admin_marketing_recommendations_specs`. `wc_marketing_recommended_plugins` should not exist.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

</details>
